SU2_CFD/src/numerics/heat.cpp

```diff
@@ -91,20 +87,21 @@ void CCentSca_Heat::ComputeResidual(su2double *val_residual, su2double **val_Jac
 
   /*--- Adding artificial dissipation to stabilize the centered scheme ---*/
 
-  Diff_Lapl[0] = Und_Lapl_i[0]-Und_Lapl_j[0];
+  su2double Diff_Lapl = Und_Lapl_i[0]-Und_Lapl_j[0];
+  su2double Area2 = GeometryToolbox::SquaredNorm(nDim, Normal);
 
-  SoundSpeed_i = sqrt(ProjVelocity_i*ProjVelocity_i + (BetaInc2_i/DensityInc_i)*Area*Area);
-  SoundSpeed_j = sqrt(ProjVelocity_j*ProjVelocity_j + (BetaInc2_j/DensityInc_j)*Area*Area);
+  su2double SoundSpeed_i = sqrt(pow(ProjVelocity_i,2) + (BetaInc2_i/DensityInc_i)*Area2);
```

Comment:
Just for my own knowledge, is pow(x,2) superior to x*x?

Reply:
I think it is more expressive, saves you the trouble of reading two variable names to make sure they are the same.
In terms of performance, the pow function is horrible, however, pow(x,2) always gets optimized to x*x.
Larger integer powers are only optimized in that way if some -ffast-math optimizations are allowed, since e.g. x*x*x is not a strict way to compute pow(x,3) (as per the floating-point standard).
That type of optimization is fairly innocuous in double-precision, the intel compilers actually do them by default.